astropy/coordinates/sky_coordinate.py

```diff
@@ -65,19 +66,35 @@ def _repr_data(self):
         return repr_data
 
     def _represent_as_dict(self):
-        obj = self._parent
-        attrs = (list(obj.representation_component_names) +
-                 list(frame_transform_graph.frame_attributes.keys()))
+        sc = self._parent
+        attrs = list(sc.representation_component_names)
+
+        # Don't output distance unless it's actually distance.
+        if issubclass(sc.data.__class__, UnitSphericalRepresentation):
+            attrs = attrs[:-1]
+
+        diff = sc.data.differentials.get('s')
+        if diff:
```

Comment:
Be careful with assuming `bool(diff)` works - this will break or at least lead to a `DeprecationWarning` for non-scalar representations. Best just to be explicit and use `if diff is not None:`

Reply:
Done.